### ISSUE TITLE
Options programmatically added/removed to/from optgroups are not rendered

### ISSUE THREAD

**Issue by augmt** [select2/select2#3604]
In [this](http://jsfiddle.net/z0hhkbg0/) fiddle I have two selects: one with just an option and another with an option within an optgroup.

After running the fiddle, the select with just an option has three options, as expected, but the select with the optgroup only has two. It seems as if the select with the optgroup does not render any newly appended options once the select has been opened.

Am I doing something unintended or is this a bug?

**Comment by gamut-was-taken** [comment]
I found that this "bug" occurs because data gets saved onto each optgroup only when the select is opened. [This](https://github.com/select2/select2/blob/master/src/js/select2/data/select.js#L205) line disallows elements with data saved onto them from updating, so an optgroup that previously had two children and now has three children would only render the two children from before.

I fixed this by appending `(!$option.is('optgroup') || $option.children('option').length === data.children.length)` onto the if condition, but I'm not sure if this is intended behavior.
